src/transactions/types.ts

```diff
@@ -301,3 +302,15 @@ export type InputGenerateTransactionData =
   | InputGenerateMultiAgentRawTransactionData
   | InputGenerateFeePayerRawTransactionData
   | InputGenerateSingleSignerRawTransactionData;
+
+/**
+ * Interface that holds the user data input when submitting a transaction
+ */
+export interface InputSubmitTransactionData {
+  transaction: AnyRawTransaction;
+  senderAuthenticator: AccountAuthenticator;
+  secondarySignerAuthenticators?: {
+    feePayerAuthenticator?: AccountAuthenticator;
+    additionalSignersAuthenticators?: Array<AccountAuthenticator>;
+  };
```

Comment:
This is a little weird now that i'm looking at it that, secondary signers, contains fee payer and additional signers.

We may want to flatten this to two top level optionals.  (Or rename the outside additional, and the internal one secondary which matches the API)

Reply:
agreed. ill flatten it